### ISSUE TITLE
Angular v16 projects are unable to use angular-instantsearch

### ISSUE THREAD

**Issue by kanehjeong** [algolia/angular-instantsearch#994]
**Describe the bug 🐛**
Angular v16 was just released. It comes with a ton of new/updated features, but projects utilizing angular-instantsearch are unable to upgrade to it.

**To Reproduce 🔍**

1. Create a new Angular v16 project, or migrate an existing project to Angular v16.
2. Start project using angular-instant-search
3. Gets error due to library not being Ivy compatible
 
**Expected behavior 💭**
- No error, able to use package as usual

**Screenshots 🖥**
<img width="1671" alt="Screenshot 2023-05-04 at 5 07 01 PM" src="https://user-images.githubusercontent.com/3455022/236355317-3ffdea9a-547d-4321-8629-1016350a7cae.png">

**Environment:**

 - OS: Mac
 - Browser: Chrome
 - Version: 4.4.1

**Comment by NateMay** [comment]
Yeah, this is gonna be a big issue for all Angular developers looking to upgrade.

**Comment by NateMay** [comment]
For all who follow, I dug into this repository and came to the following conclusion: This package has few weekly downloads, it will be a nightmare to update, and even if and when the team decides to upgrade it will have many breaking changes. If I were in charge of this project, I would probably decide not to update this library, and opt instead to add a brand new angular library to the roadmap.

**Comment by kanehjeong** [comment]
@NateMay I actually came to the same conclusion soon after I posted this ticket since this project didn't seem that actively maintained and luckily we don't utilize algolia that heavily yet (we have 1 index).

I decided to keep this ticket open since it is an actual issue to be addressed (if ever). But for anyone looking to not be blocked, highly suggest what me and @NateMay did. The above is a good starting point for sure :)

**Comment by imbradyboy** [comment]
@NateMay curious to hear more about your use of algolia as I'm running into a similar issue. My hesitance around using the vanilla Algolia search library is that we require semi-complex querying on our index (eg. search, date ranges, refinement lists, and location-based search).

Our team has been extending the provided widgets to make all of these filters work together, but we've gotten to a point where the GeoSearch implementation for angular no longer fits our needs and is buggy.

My main question for you is, are you just using full-text search? Do you see yourself using more filters in the future?

**Comment by NateMay** [comment]
>are you just using full-text search? Do you see yourself using more filters in the future?

I was extending widgets too, and I am (and was) using filters and facet searching, but no GeoSearch. 

As you can see in the service that I provided above, you can pass a SearchResponse object. The angular-instantsearch library managed this in the template but ultimately passed those options to the [algoliasearch](https://www.npmjs.com/package/algoliasearch) library. You'll have to learn the API, but there is nothing that can't be done since the former relies on the latter. 

In my experience, my new implementation is much cleaner.

**Comment by imbradyboy** [comment]
> > are you just using full-text search? Do you see yourself using more filters in the future?
> 
> I was extending widgets too, and I am (and was) using filters and facet searching, but no GeoSearch.
> 
> As you can see in the service that I provided above, you can pass a SearchResponse object. The angular-instantsearch library managed this in the template but ultimately passed those options to the [algoliasearch](https://www.npmjs.com/package/algoliasearch) library. You'll have to learn the API, but there is nothing that can't be done since the former relies on the latter.
> 
> In my experience, my new implementation is much cleaner.

Yep, fair points. Thanks for the insight! We've been hesitant to go all in on the instantsearch library and invest that time, but seems that it may be our best option going forward. Appreciate you taking the time to respond!

**Comment by johan--** [comment]
Any ETA on adding Ivy compiler support?  We are using this in an Angular app, and cannot upgrade to Angular 16 because of a dependency on this library.

**Comment by gethari** [comment]
Hi Team, any update or workaround on this ?

**Comment by gethari** [comment]
Related to #1003

**Comment by loveraimogen** [comment]
We're moving the discussion over to an Algolia owned [Issue #5924](https://github.com/algolia/instantsearch/issues/5924). Updates on this topic will be posted there.
